### PR TITLE
Astractl 27293 updates for releae branch

### PR DESCRIPTION
**Pull request by jharrod**
Updates versioning/tagging in connector operator repo.

Versioning file now has all three digits - X.Y.Z

For release branches versions and tags look like
- "1.2.3-202305102210"

For non-release branches:
BUILD-BRANCH (BUILD is currently a timestamp)

### CODE REVIEW

File: scripts/create-image-tar.sh [hunk 1]
```diff
@@ -28,7 +28,7 @@ done
 
 # Get operator-image
 operatorYamlPath="${parentDir}/../astraconnector_operator.yaml"
-pattern=' +image: netapp\/astra-connector-operator:([^\n ]*)'
```
Comment (jharrod) [inline scripts/create-image-tar.sh:31]: Not sure why this wasn't working, it was cutting off the end of my version when it was in the BUILD-BRANCH form

File: Makefile [hunk 2]
```diff
@@ -112,10 +112,7 @@ docker-build: ## Build docker image with the manager.
 
 docker-push: ## Push docker image with the manager.
 	docker push ${IMG}
-docker-tag-base:
-	docker tag 	${IMG} ${IMAGE_TAG_BASE}:${BASE_VERSION}
-docker-push-base:
```
Comment (jharrod) [inline Makefile:117]: These were not used as far as I could tell, and were confusing, so I removed them

**Review by raghu-sarangapani — APPROVED** [review]

**Comment by josecrosa** [comment]
typo in PR title: releae > release

File: version.txt [hunk 3]
```diff
@@ -1 +1 @@
-2.0
\ No newline at end of file
+2.1.0
```
Comment (bmason42) [inline version.txt:1]: We need to pop this to 3.0.0
We already have a 2 series and this is a breakingish change

**Review by bmason42 — CHANGES_REQUESTED** [review]

**Review by bmason42 — APPROVED** [review]

**Pull request merged**